File: algorithms/io/IO.java
/*
 * Classe para operações de I/O em console utilizando a classe Scanner para
 * a entrada.
 *
 * Autor: Edson Alves
 * Data: 18/02/2015
 * Licença: LGPL. Sem copyright.
 */
package algorithms.io;

import java.util.Scanner;
import java.util.ArrayList;

public class IO {

    private Scanner sc = new Scanner(System.in);

    public final int[] read() {
        ArrayList<Integer> list = new ArrayList<Integer>();

        while (sc.hasNext()) {
            int n = sc.nextInt();
            list.add(n);
        }

        int[] ns = new int[list.size()];

        for (int i = 0; i < ns.length; i++) {
            ns[i] = list.get(i).intValue();
        }

        return ns;
    }

    public final void write(final int[] ns) {
        for (int i : ns) {
            System.out.println("" + i);
        }
    }
}

File: tests/io/io.cpp
/*
 * Classe para teste de execução do método echo() da classe IO.
 *
 * Autor: Edson Alves
 * Data: 12/03/2015
 * Licença: LGPL. Sem copyright.
 */
#include <algorithms/io/io.h>

#include <chrono>

using namespace std;

int main()
{
    auto start = chrono::system_clock::now();
    auto ns = IO<int>::read();
    auto end = chrono::system_clock::now();
    chrono::duration<double> rtime = end - start;
 
    start = chrono::system_clock::now();
    IO<int>::write(ns);
    end = chrono::system_clock::now();
    chrono::duration<double> wtime = end - start;

    cout.precision(6);
    cout << rtime.count() << "ms to read " << ns.size() << " integers\n";

    cout.precision(6);
    cout << wtime.count() << "ms to write " << ns.size() << " integers\n";

    return 0;
}
